tests/phpunit/Persistence/Search/SearchIndexFieldsBuilderTest.php

```diff
@@ -30,10 +30,12 @@ public function setUp(): void {
 		$this->cirrusSearch = new CirrusSearch();
 	}
 
-	public function testEmptyConfigReturnsNoFields(): void {
+	public function testEmptyConfigThrowsException(): void {
 		$builder = $this->newBuilder( new Config() );
 
-		$this->assertSame( [], $builder->createFields() );
+		$this->expectException( \RuntimeException::class );
+
+		$builder->createFieldObjects();
```

Comment:
I imagine to avoid this type of test where practically it should not happen (i.e. we're not constructing the object if Config is incomplete), but technically it can, we might have to inject the known values into SearchIndexFieldsBuilder instead of the whole Config object.